internal/core/labels.go

```diff
@@ -17,12 +18,26 @@ const (
 	LabelVersion   = LabelBase + ".version"
 )
 
+// DefaultLabels returns the standard set of labels which
+// includes LabelSessionID if the reaper is enabled.
 func DefaultLabels(sessionID string) map[string]string {
-	return map[string]string{
-		LabelBase:      "true",
-		LabelLang:      "go",
-		LabelSessionID: sessionID,
-		LabelVersion:   internal.Version,
+	labels := map[string]string{
+		LabelBase:    "true",
+		LabelLang:    "go",
+		LabelVersion: internal.Version,
+	}
+
+	if !config.Read().RyukDisabled {
+		labels[LabelSessionID] = sessionID
```

Comment:
I'd not remove the session ID, as this label could be used in the backend to connect test sessions with remote VM executions.

Reply:
We have to exclude it otherwise the reaper would attempt to clean it when the configuration say specifically not to.

Reply:
switched to a dedicated reap label.